### PR TITLE
Strip warehouse name in mutations

### LINKED ISSUE #5673: Warehouse name is not stripped in validation
User can create a warehouse without a name by passing empty spaces as the name in API.

### Steps to reproduce the problem
1. Run this mutation:
```
mutation {
  createWarehouse(
    input: {
      name: " "
      slug: "test-12345"
      address: {
        country: PL
        city: "Test"
        streetAddress1: "Test"
        postalCode: "12-345"
      }
    }
  ) {
    warehouseErrors {
      field
      code
    }
    warehouse {
      id
      name
      slug
    }
  }
}
```
Note that `name` is ` `.

2. Mutation creates a warehouse with an empty name.

### What I expected to happen
I should get the error:
```
{
    "field": "name",
    "code": "REQUIRED"
},
```

### PR DESCRIPTION
**Pull request by koradon**
I want to merge this change because...

fixes #5673 

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [x] Privileged queries and mutations are guarded by proper permission checks
* [x] Database queries are optimized and the number of queries is constant
* [x] Database migration files are up to date
* [x] The changes are tested
* [x] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/warehouse/tests/test_warehouse.py [hunk 1]
```diff
@@ -374,6 +374,42 @@ def test_mutation_create_warehouse(
     assert created_warehouse["slug"] == warehouse.slug
 
 
+def test_mutation_create_warehouse_does_not_create_when_name_is_empty_string(
+    staff_api_client, permission_manage_products, shipping_zone
+):
+    Warehouse.objects.all().delete()
```
Comment (fowczarek) [inline saleor/graphql/warehouse/tests/test_warehouse.py:380]: Why are you removing all the warehouses?

Comment (koradon) [reply saleor/graphql/warehouse/tests/test_warehouse.py:380]: Honestly, I copied the test above and changed just what I needed. Missed that line.

File: saleor/graphql/core/utils/__init__.py [hunk 2]
```diff
@@ -108,3 +108,15 @@ def get_duplicates_ids(first_list, second_list):
 def get_duplicated_values(values):
     """Return set of duplicated values."""
     return {value for value in values if values.count(value) > 1}
+
+
+def validate_and_strip_field_str_value(cleaned_input, field_name: str):
```
Comment (maarcingebala) [inline saleor/graphql/core/utils/__init__.py:113]: A detail, but maybe `validate_required_string_field` would sound more universal?

Comment (koradon) [reply saleor/graphql/core/utils/__init__.py:113]: Yes, I was thinking about it but it's not exactly the case here. Field `name` is required to create mutation and optional for the update.
`not_empty` or other combinations are also not the best IMO. I tried to connect validation and the exact action here. But I am not happy either with the result.
(Naming is hard sometimes)

Comment (maarcingebala) [reply saleor/graphql/core/utils/__init__.py:113]: The difference between updating and creating is handled by the schema where the field is required or not. Another option is to move `if field_name in cleaned_input` outside and in the function assume that `cleaned_input["value"]` is present? 

We have the same logic in `OrderAddNote` and we should reuse this function there as well.

Comment (koradon) [reply saleor/graphql/core/utils/__init__.py:113]: Ok I will reuse it there.

**Pull request merged**